crates/proto/src/clients/mod.rs

```diff
@@ -0,0 +1,633 @@
+            .map_err(|e| ClientError::InvalidMetadata(e.to_string()))?;
+
+        let mut metadata = HashMap::new();
+        metadata.insert("accept", metadata_value);
+
+        Ok(MetadataInterceptor { metadata })
+    }
+}
+
+// METADATA INTERCEPTOR
+// ================================================================================================
+
+/// Interceptor for adding metadata to RPC requests.
+#[derive(Clone)]
+pub struct MetadataInterceptor {
+    metadata: HashMap<&'static str, AsciiMetadataValue>,
+}
+
+impl Interceptor for MetadataInterceptor {
+    fn call(
+        &mut self,
+        mut request: tonic::Request<()>,
+    ) -> Result<tonic::Request<()>, tonic::Status> {
+        for (key, value) in &self.metadata {
+            request.metadata_mut().insert(*key, value.clone());
+        }
+        Ok(request)
+    }
+}
+
+// TYPE ALIASES
+// ================================================================================================
+
+/// Type alias for RPC API client with metadata interceptor.
+pub type RpcApiClient =
+    crate::generated::rpc::api_client::ApiClient<InterceptedService<Channel, MetadataInterceptor>>;
+
+/// Type alias for Block Producer API client with OpenTelemetry interceptor.
+pub type BlockProducerApiClient = crate::generated::block_producer::api_client::ApiClient<
+    InterceptedService<Channel, OtelInterceptor>,
+>;
+
+// CLIENT WRAPPERS
+// ================================================================================================
+
+/// RPC store client wrapper with OpenTelemetry tracing.
+///
+/// This wrapper provides access to all RPC methods like `check_nullifiers`,
+/// `get_account_details`, `sync_notes`, etc.
+#[derive(Debug)]
+pub struct RpcStoreClient {
+    inner: crate::generated::store::rpc_client::RpcClient<
+        InterceptedService<Channel, OtelInterceptor>,
+    >,
+}
+
+impl RpcStoreClient {
+    fn new(channel: Channel) -> Self {
+        let inner = crate::generated::store::rpc_client::RpcClient::with_interceptor(
+            channel,
+            OtelInterceptor,
+        );
+        Self { inner }
+    }
+
+    /// Get a mutable reference to the underlying client.
+    pub fn get_mut(
+        &mut self,
+    ) -> &mut crate::generated::store::rpc_client::RpcClient<
+        InterceptedService<Channel, OtelInterceptor>,
+    > {
+        &mut self.inner
+    }
+
+    /// Get the underlying client.
+    pub fn into_inner(
+        self,
+    ) -> crate::generated::store::rpc_client::RpcClient<InterceptedService<Channel, OtelInterceptor>>
+    {
+        self.inner
+    }
+}
+
+impl Clone for RpcStoreClient {
+    fn clone(&self) -> Self {
+        Self { inner: self.inner.clone() }
+    }
+}
+
+/// `BlockProducer` store client wrapper with OpenTelemetry tracing.
+///
+/// This wrapper provides access to all `BlockProducer` methods like `get_transaction_inputs`,
+/// `get_block_inputs`, `apply_block`, etc.
+#[derive(Debug)]
+pub struct BlockProducerStoreClient {
```

Comment:
Seems like these are pretty much the same (`BlockProducerStoreClient`, `BlockProducerStoreClient`, `RpcStoreClient`), is there any benefit to making these explicit instead of having a more generic `build<T>` as suggested [here](https://github.com/0xMiden/miden-node/issues/869#issuecomment-3022224233)?